onnx/defs/controlflow/defs.cc

```diff
@@ -5,6 +5,171 @@
 namespace ONNX_NAMESPACE {
 using SupportType = OpSchema::SupportType;
 
+static void UpdateValueFromDim(
+    int64_t& current_value,
+    const TensorShapeProto_Dimension& dim) {
+  if (current_value == -1 && dim.has_dim_value()) {
+    auto value = dim.dim_value();
+    if (value != -1) {
```

Comment:
Can we drop this check? It is not really needed here. (It is better if we adopt one uniform scheme to indicate an unnamed, unknown, dimension … I believe that a dimension that has no dim_value and no dim_param is used for this purpose currently. Is there any usage of dim_value == -1 for this purpose also?)

Reply:
This is based on the initialization value for the local variable rather than expecting to see -1 as dim_value. 
  int64_t batch_size = -1;
  int64_t sequence_len = -1;

Primary purpose is to know if a real batch size or sequence length value was seen. 
Would current_value < 1 be fine? 
Should I use something other than -1 as the 'not set' value?

Reply:
Isn't that the outer check for "current_value" (that you are referring to)? This "value" is dim.dim_value() ...

Reply:
Ah yes - sorry - the inner check can go. Will remove.